compiler/rustc_typeck/src/check/_match.rs

```diff
@@ -188,11 +188,18 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
             } else {
-                let (arm_span, semi_span) = if let hir::ExprKind::Block(blk, _) = &arm.body.kind {
```

Comment:
In the spirit of keeping the happy path clean, can you move all the span and obligation cause computation here into a function?